libusb/os/threads_posix.h

```diff
@@ -51,7 +51,8 @@ static inline void usbi_mutex_unlock(usbi_mutex_t *mutex)
 }
 static inline int usbi_mutex_trylock(usbi_mutex_t *mutex)
 {
-	return pthread_mutex_trylock(mutex) == 0;
+	int mutexIsNowLocked = pthread_mutex_trylock(mutex) == 0;
```

Comment:
I am fine with this way of doing, though I am not used to and and I think a comment could have the same informative effect. 

I think the risk with this is that later at some point someone will just think the temporary variable is not necessary (and this is indeed the case, technically) and remove it for "cleaning" the code. 

I would prefer a comment but I can live with this.